#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()

gen.add("use_x_dimension", bool_t, 0, "Use x dimension to filter", False)
gen.add("x_dimension_min", double_t, 0, "Minimum value for x dimension", 0.1, 0.0, 10.0)
gen.add("x_dimension_max", double_t, 0, "Maximum value for x dimension", 0.1, 0.0, 10.0)
gen.add("use_y_dimension", bool_t, 0, "Use y dimension to filter", False)
gen.add("y_dimension_min", double_t, 0, "Minimum value for y dimension", 0.1, 0.0, 10.0)
gen.add("y_dimension_max", double_t, 0, "Maximum value for y dimension", 0.1, 0.0, 10.0)
gen.add("use_z_dimension", bool_t, 0, "Use z dimension to filter", False)
gen.add("z_dimension_min", double_t, 0, "Minimum value for z dimension", 0.1, 0.0, 10.0)
gen.add("z_dimension_max", double_t, 0, "Maximum value for z dimension", 0.1, 0.0, 10.0)
gen.add("filter_limit_negative", bool_t, 0, "Set to true if we want to return the data outside [filter_limit_min; filter_limit_max].", False)
exit (gen.generate (PACKAGE, "jsk_pcl_ros", "BoundingBoxFilter"))
